R/model-persistence.R

```diff
@@ -243,6 +260,9 @@ unserialize_model <- function(model, custom_objects = NULL, compile = TRUE) {
 export_savedmodel.keras.engine.training.Model <- function(object, export_dir_base, ...) {
   if (!is_backend("tensorflow"))
     stop("'export_savedmodel' is only supported in the TensorFlow backend.")
+
+  k_set_learning_phase(FALSE)
```

Comment:
Maybe see if we can roll back the setting of the learning phase after the function (maybe check it's value beforehand?). Just worried about potential side effects. If there is no way to restore it to a previous "unset" value I still think this is a good change.

Reply:
It looks like it does support an "unset" value (a tensor that keras automatically feeds the right thing). So I think all you need to do is:

```r
old_learning_phase <- k_learning_phase()
on.exit(k_set_learning_phase(old_learning_phase), add = TRUE)
k_set_learning_phase(0)
```